lightning/src/ln/channelmanager.rs

```diff
@@ -2135,13 +2150,28 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 				});
 			},
 			HTLCSource::PreviousHopData(hop_data) => {
+				let hop_data_node_id = hop_data.counterparty_node_id;
+				let outpoint = hop_data.outpoint;
 				if let Err((counterparty_node_id, err)) = match self.claim_funds_from_hop(&mut channel_state_lock, hop_data, payment_preimage) {
 					Ok(()) => Ok(()),
 					Err(None) => {
-						// TODO: There is probably a channel monitor somewhere that needs to
-						// learn the preimage as the channel already hit the chain and that's
-						// why it's missing.
-						Ok(())
+						let preimage_update = ChannelMonitorUpdate {
+							update_id: std::u64::MAX,
```

Comment:
Corrected the docs and added an assertion that we only accept this sentinel on `payment_preimage` monitor update.
> Isn't this imply that previous link's ChannelMonitor.latest_update_id must be set up to std::u64::MAX when we don't expect anymore cooperative-update-from-offchain to avoid discrepancy for this PaymentPreimage update ?

Hmm, what action item are you thinking for this? not sure I follow..

Reply:
Actually could you replace std::u64::MAX with a named constant like `WILDCARD_ID_FROM_SWITCH`. 

It would underscore better that this update has been triggered by a different "source" than the Channel. Namely the ChannelManager when switching backward the HTLC at `claim_funds_internal`.

Reading this https://en.wikipedia.org/wiki/Sentinel_value, I wouldn't consider this a _pure_ sentinel value, we may have multiple such PaymentPreimage updates. It's not a terminal value but more a wildcard to uplift normal checks.

> Hmm, what action item are you thinking for this?

I did have a doubt on the following scenario triggering the panic L1233 in `src/chain/channelmonitor.rs`.

* ChannelMonitor receives a PaymentPreimage update from forward link, `latest_update_id` sets to std::u64::MAX
* ChannelMonitor receives a LatestCounterpartyCommitmentTXInfo, its `updates_id` is different from std::u64::MAX, thus verifying the condition L1232.

BUT, this can't happen as a PaymentPreimage-with-sentinel will only be generated if Channel can't issue anymore updates due to have been force-closed.

I think the new assertion is okay for now, I hope to simplify once we rationalize our usage of monitor state flags (`lockdown_from_offchain`/`holder_commitment_signed`).

Reply:
Updated to a named `const`!